intro-to-object-oriented-programming/practice-intro-to-oop.md

```diff
@@ -8,16 +8,866 @@ Complete all questions below.
+### !tests
+```python
+import unittest
+from main import *
+
+class TestChallenge(unittest.TestCase):
+    def test_pokemon(self):
+        gengar = Pokemon("Gengar", "ghost", "poison")
+        self.assertEqual(gengar.name, "Gengar")
+        self.assertEqual(gengar.primary_type, "ghost")
+        self.assertEqual(gengar.secondary_type, "poison")
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+class Pokemon:
+
+    def __init__(self, name, primary_type, secondary_type):
+        self.name = name
+        self.primary_type = primary_type
+        self.secondary_type = secondary_type
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 6 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: ordering
+* id: HYSC7n
+* title: Intro to OOP
+##### !question
+
+Read the following prompts. Then, arrange the code snippets below so they match the order of the prompts.
+
+1. Creates an instance of the `Pokemon` class
+1. Reads the instance of the `Pokemon` class
+1. Accesses the attribute `name` from the instance of `Pokemon`
+1. Accesses the attribute `primary_type` from the instance of `Pokemon`
+1. Accesses the attribute `secondary_type` from the instance of `Pokemon`
+1. Prints the attribute `name` from the instance of `Pokemon`
+
+##### !end-question
+##### !answer
+
+1. `tangusaur = Pokemon("Tangusaur", "poison", "grass")`
+1. `tangusaur`
+1. `tangusaur.name`
+1. `tangusaur.primary_type`
+1. `tangusaur.secondary_type`
+1. `print(tangusaur.name)`
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### !callout-secondary
+
+## Tangusaur?
+
+The fictional Pokemon in this problem set were [randomly generated](https://pokemon.alexonsager.net/).
+
+### !end-callout
+
+<!-- Question 7 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: ordering
+* id: betEim
+* title: Intro to OOP
+##### !question
+
+Assume this definition of the `Pokemon` class. Note that it uses the keyword argument syntax.
+
+```python
+class Pokemon:
+
+    def __init__(self, name, primary_type="normal", secondary_type=None):
+        self.name = name
+        self.primary_type = primary_type
+        self.secondary_type = secondary_type
+```
+
+Read the following code snippets. Then, arrange the options below so they match the order of the code snippets.
+
+1. `abchan = Pokemon("Abchan", primary_type="psychic", secondary_type="fighting")`
+1. `magibat = Pokemon("Magibat", primary_type="water", secondary_type="flying")`
+1. `"Abchan"`
+1. The instance of the Pokemon with a `name` of `"Abchan"`
+1. `"flying"`
+1. `"water"`
+1. `"psychic"`
+1. `"fighting"`
+
+##### !end-question
+##### !answer
+
+1. Creates an instance of `Pokemon` where the `primary_type` has a value of `psychic` and the `secondary_type` has a value of `fighting`
+1. Creates an instance of `Pokemon` where the `primary_type` has a value of `water` and the `secondary_type` has a value of `flying`
```

Comment:
Would love to hear if there's pushback on this because of clarity.

I happen to really like the exercise of it going this direction: Forcing a student to read through a line of code, verbalize it into their own head, read someone else's explanation of it, and then match it.

Otherwise, personally I could see students resistant to reading code glazing their eyes over and trying to match keywords as quickly as possible (visually matching `"flighting"` to `"fighting"`). I get that's.... possible in this direction too... :P BUT IT'S DIFFERENT!!!!!

Reply:
Also I just revisited this.... I totally get what you mean about code in one section, and descriptions in another. Let me rethink it...

Reply:
Then I would suggest flipping the other options. The problem does say that the student is to read the "code snippets" and then order the options. As it currently is, the first two prompts _are_ code snippets, but everything else is the result of _evaluating_ a code snippet.